# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2004/12/24 22:33:13-08:00 sri@us.ibm.com 
#   [SCTP] Fix potential null pointer dereference in sctp_err_lookup().
#   
#   Signed-off-by: Vladislav Yasevich <vladislav.yasevich@hp.com>
#   Signed-off-by: Sridhar Samudrala <sri@us.ibm.com>
# 
# net/sctp/input.c
#   2004/12/24 22:33:04-08:00 sri@us.ibm.com +2 -1
#   [SCTP] Fix potential null pointer dereference in sctp_err_lookup().
# 
diff -Nru a/net/sctp/input.c b/net/sctp/input.c
--- a/net/sctp/input.c	2005-01-15 13:40:55 -08:00
+++ b/net/sctp/input.c	2005-01-15 13:40:55 -08:00
@@ -326,11 +326,12 @@
 	}
 
 	if (asoc) {
+		sk = asoc->base.sk;
+
 		if (ntohl(sctphdr->vtag) != asoc->c.peer_vtag) {
 			ICMP_INC_STATS_BH(ICMP_MIB_INERRORS);
 			goto out;
 		}
-		sk = asoc->base.sk;
 	} else
 		sk = ep->base.sk;
 
